runtime/fiber.c

```diff
@@ -583,3 +592,14 @@ CAMLprim value caml_drop_continuation (value cont)
   caml_free_stack(stk);
   return Val_unit;
 }
+
+CAMLprim value caml_fiber_id(value unit)
+{
+  struct stack_info* stack = Caml_state->current_stack;
+
+  CAMLassert(unit == Val_unit);
+  if (!stack || stack->id > INT_MAX)
```

Comment:
This should probably be improved: `id` is an `uintnat` to get the free overflow handling by C but conversion to OCaml `int` will overflow sooner!